commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/gitlab/GitlabPublisher.java

```diff
@@ -98,6 +107,28 @@ private void publish(@NotNull SBuild build,
       throw new PublisherException("Cannot parse repository URL from VCS root " + root.getName());
 
     String message = createMessage(status, build, revision, myLinks.getViewResultsUrl(build), description);
+
+    Pattern r = Pattern.compile("/merge-requests/([0-9]+)/");
+    Matcher m = r.matcher(revision.getRepositoryVersion().getVcsBranch());
+    if (m.find( )) {
```

Comment:
This is minor, but worth noticing: GitLab branch name may contain slashes, so it seems using find() you in theory may match a branch name that is not a merge request, but its name just contain something that is matched by the pattern.